src/test/java/com/google/sps/data/ShoppingCartHandlerTest.java

```diff
@@ -0,0 +1,69 @@
+package com.google.sps.data;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.tools.development.testing.LocalDatastoreServiceTestConfig;
+import com.google.appengine.tools.development.testing.LocalServiceTestHelper;
+import java.util.ArrayList;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+
+@RunWith(JUnit4.class)
+public class ShoppingCartHandlerTest {
+  private final LocalServiceTestHelper helper = new LocalServiceTestHelper(new LocalDatastoreServiceTestConfig());
+  private ShoppingCartHandler cartHandler = new ShoppingCartHandler();
+
+  @Before
+  public void setUp() {
+    helper.setUp();
+  }
+
+  @After
+  public void tearDown() {
+    helper.tearDown();
+  }
+
+  @Test
+  public void testGettingShoppingCart() {
```

Comment:
For each of these tests add another one which checks that everything works for a user without a shopping cart.

Reply:
In the tests in the beginning users do not have a UserCart in the datastore and that case is tested.

Reply:
Not for all the tests in this file (get/add/delete).

Also empty shopping cart should be its own test. In general you want to keep tests super small and focused, so don't bundle them together (i.e testGettingShoppingCartWorksForNewBasket, testGettingShoppingCartWorksForExistingBasket).